### PR TITLE
Fixed #138 + Minor update to ReadMe

### LINKED ISSUE #138: 404 when clicking SEARCH directly on homepage without input
On the homepage, when we click on search🔍 button directly, without providing any input text or providing just blank spaces takes the user to a 404 page.

**Expected**: Prompt user to provide input
**Actual**: Considers empty field as input and couldn't perform a proper search

If this issue is concerning, allow me to fix it.

image

### PR DESCRIPTION
**Pull request by prithvi2k2**
- Added `required` attribute to input field which now won't allow users to search with empty input and thus, won't reroute to 404 page. Fixes #138 

- Generalized Readme.md, it mentioned Hacktoberfest **2022** but the site is dynamic and works well in long-term, so I removed "**2022**".

### CODE REVIEW

**Comment by max-programming** [comment]
Thanks! Can you also add a validation that if the search value is empty, it won't proceed

**Comment by prithvi2k2** [comment]
@max-programming my bad, I honestly thought `required` attribute will check for empty search values(like spaces only)... 
 updated code as follows:

- Will show the following error with `tailwind : animate-pulse` if an invalid input is entered - No empty input or no spaces only input - both aren't allowed. 
image
- I validated it by trimming leading & trailing spaces of input and compared it with empty string. Hope this fixes the issue.
-  Also, removed the `required` attribute as `handleSubmit()` is checking for the same.

Please tell, if something still doesn't feel right.

**Comment by prithvi2k2** [comment]
@max-programming Is this PR worth the `hacktoberfest-accepted` label? I'll make more contributions soon...

**Comment by max-programming** [comment]
@prithvi2k2 Don't worry about it getting accepted in hacktoberfest. Once it is merged, it will be approved

**Comment by prithvi2k2** [comment]
@max-programming Resolved a merge conflict with `Readme.md` file, hopefully, this can be merged now without a problem

**Review by max-programming — CHANGES_REQUESTED** [review]

File: components/Hero.tsx [hunk 1]
```diff
@@ -10,11 +10,20 @@ import Button from './Button';
...
   const handleSubmit: FormEventHandler = e => {
     e.preventDefault();
     const formData = new FormData(e.target as HTMLFormElement);
-    const search = formData.get('search');
+    const search = (formData.get('search') as string).trim();
+    // Check if the input is empty or contains only spaces
+    if (search === '') {
+      setErrorMessage('Empty search terms invalid!');
+      return;
+    } else {
```
[the remark this answers is not in the document]
Comment (prithvi2k2) [reply components/Hero.tsx:23]: Yeah, sure, I didn't refactor it after testing... my bad

[the remark this answers is not in the document]
Comment (prithvi2k2) [reply components/Hero.tsx]: Updated with commit [65e8759](https://github.com/max-programming/hacktoberfest-projects/pull/142/commits/65e8759413aa76e49b626ebc0292e399255c2c45)

**Comment by max-programming** [comment]
Alright @prithvi2k2 
But the error message does not show up when I hit search with empty. Can you fix that?
Also, make sure to use one of the red colors in the tailwind config so that it matches the theme

**Comment by max-programming** [comment]
And can you do the same for the searchbox on other pages?
image

**Review by max-programming — APPROVED** [review]

**Comment by prithvi2k2** [comment]
Yes, the search other pages already doesn't take empty input... but for other inputs, I need to trim leading and trailing spaces to make it good... will try to do that

**Comment by prithvi2k2** [comment]
EXTRA: Found that search icon on home page when **hovered** is ugly due to high opacity hover bg color. Decreased the opacity to make it look good.

Before : 
High Opacity BG


After: 
Low Opacity BG

**Comment by prithvi2k2** [comment]
> EXTRA: Found that search icon on home page when **hovered** is ugly due to high opacity hover bg color. Decreased the opacity to make it look good.
> 
> Before : High Opacity BG
> 
> After: Low Opacity BG

Reverted the search icon change commit #c3a37d8 - as it's already in progress at #170 & #174 ... I didn't check mb. I shouldn't have been fixing a different issue in this PR, will not repeat that and try to create issue specific PRs only.

File: components/Search.tsx [hunk 2]
```diff
@@ -40,7 +43,7 @@ export default function Search({ searchBarWrapperStyles }: SearchProps) {
             type="button"
             onClick={() => reset()}
           >
-            <GoX />
+            <GoX color='white'/>
```
Comment (max-programming) [inline components/Search.tsx:46]: Can you revert this because it's already fixed now?

File: components/Search.tsx [hunk 3]
```diff
@@ -19,7 +19,10 @@ export default function Search({ searchBarWrapperStyles }: SearchProps) {
   });
 
   const onSubmit: SubmitHandler<FormValues> = ({ searchQuery }) => {
-    router.push({ query: { ...router.query, q: searchQuery } });
+    let trimmedQuery = searchQuery.trim();
+    if(trimmedQuery !== ''){ //Performs search only with non-empty strings
```
Comment (max-programming) [inline components/Search.tsx:23]: Just run prettier to format the code now

**Review by max-programming — CHANGES_REQUESTED** [review]

File: components/Search.tsx (hunk 2 above)
Comment (max-programming) [reply components/Search.tsx]: I fixed it because there were conflicts

File: components/Search.tsx (hunk 3 above)
Comment (prithvi2k2) [reply components/Search.tsx]: Added commit after formatting

**Comment by prithvi2k2** [comment]
Formatted with prettier.... And regarding the **clear input button** colour... please test it on light mode... I gave it white colour as it wasn't visible in light mode. You can redact it if it's unnecessary or tell me if I should revert that change... I didn't revert it now as it was marked resolved

**Comment by max-programming** [comment]
> Formatted with prettier.... And regarding the **clear input button** colour... please test it on light mode... I gave it white colour as it wasn't visible in light mode. You can redact it if it's unnecessary or tell me if I should revert that change... I didn't revert it now as it was marked resolved

There were conflicts and I fixed it

**Comment by max-programming** [comment]
@all-contributors please add @prithvi2k2 for code, design

**Comment by max-programming** [comment]
@all-contributors please add @prithvi2k2 for code, design

**Comment by prithvi2k2** [comment]
Nevermind... let it be, continue with other work @max-programming np at all not being added in contributors list

**Comment by max-programming** [comment]
@all-contributors please add @prithvi2k2 for code, design

**Comment by max-programming** [comment]
> Nevermind... let it be, continue with other work @max-programming np at all not being added in contributors list

No worries. I fixed it ✌️

**Pull request merged**